upstairs/src/block_io.rs

```diff
@@ -63,22 +63,18 @@ impl BlockIO for FileBlockIO {
     async fn read(
         &self,
         offset: Block,
-        data: Buffer,
-    ) -> Result<(), CrucibleError> {
-        let mut data_vec = data.as_vec().await;
-        let mut owned_vec = data.owned_vec().await;
-
-        let start = offset.value * self.block_size;
+        mut data: Buffer,
+    ) -> Result<Buffer, CrucibleError> {
+        let start: usize = (offset.value * self.block_size) as usize;
 
         let mut file = self.file.lock().await;
-        file.seek(SeekFrom::Start(start))?;
-        file.read_exact(&mut data_vec[..])?;
+        file.seek(SeekFrom::Start(start as u64))?;
 
-        for i in 0..data_vec.len() {
-            owned_vec[i] = true;
-        }
+        let mut buf = vec![0u8; data.len()];
+        file.read_exact(&mut buf)?;
+        data.write(0, &buf);
```

Comment:
I added ownership to `FileBlockIO` in 6a0983ad0409700df67cf072b2b2fa672e9a10c0.

For the allocation question: I don't see a good way for us to do something like

```rust
file.read_exact(&mut *data)?;
```

It would make sense to have an `impl AsMut<u8>` for Buffer _except_ that ownership information would not appropriately be filled in.

Reply:
I reverted the ownership addition - it didn't make sense because it wasn't stored in the backing `File`, and if the `FileBlockIO` was dropped and recreated the ownership would be lost. Without any sort of persistent ownership storage, it makes sense to me for it to always return `true`.

Reply:
I added a block comment about ownership in fd61bb6